### PR TITLE
Eth2-to-Near-relay: interaction with DAO

### PR DESCRIPTION
**Pull request by olga24912**
Implementation of Eth Client Wrapper for interaction with DAO.

We created a trait for Eth Client Wrapper with two implementations: (1) direct interaction with contract(eth_client_contract.rs), (2) submitting the light client updates by DAO (**dao_eth_client_contract.rs**).

In DAO ETH2 Client wrapper implementation, all functions except `send_light_client_update` work in the same way as in the classic implementation and call the correspondent functions in EthClientContract inside.  In `send_light_client_update` we submit the light client update to DAO and actively wait for when the transaction will be handled on DAO (until the status changes from `InProgress`).

### CODE REVIEW

**Comment by sept-en** [comment]
@olga24912 please rebase/merge on top of the `the-merge` branch.

**Review by karim-en — APPROVED** [review]

**Review by sept-en — APPROVED** [review]
IMO it's not the best way to wait for a proposal approval after sending a light client update, meaning the relayer will not be doing anything else at that time (submitting blocks). However, we can improve this in upcoming PR.

**Pull request merged**
